tests/integration/test_types_integration.py

```diff
@@ -0,0 +1,294 @@
+        .add_field(sql="CAST(9223 AS BIGINT)", python=9223) \
+        .add_field(sql="CAST(9223372036854775807 AS BIGINT)", python=9223372036854775807) \
+        .execute()
+
+
+def test_real(trino_connection):
+    SqlTest(trino_connection) \
+        .add_field(sql="CAST(null AS REAL)", python=None) \
+        .add_field(sql="CAST('NaN' AS REAL)", python=math.nan) \
+        .add_field(sql="CAST('-Infinity' AS REAL)", python=-math.inf) \
+        .add_field(sql="CAST(3.4028235E38 AS REAL)", python=3.4028235e+38) \
+        .add_field(sql="CAST(1.4E-45 AS REAL)", python=1.4e-45) \
+        .add_field(sql="CAST('Infinity' AS REAL)", python=math.inf) \
+        .execute()
+
+
+def test_double(trino_connection):
+    SqlTest(trino_connection) \
+        .add_field(sql="CAST(null AS DOUBLE)", python=None) \
+        .add_field(sql="CAST('NaN' AS DOUBLE)", python=math.nan) \
+        .add_field(sql="CAST('-Infinity' AS DOUBLE)", python=-math.inf) \
+        .add_field(sql="CAST(1.7976931348623157E308 AS DOUBLE)", python=1.7976931348623157e+308) \
+        .add_field(sql="CAST(4.9E-324 AS DOUBLE)", python=5e-324) \
+        .add_field(sql="CAST('Infinity' AS DOUBLE)", python=math.inf) \
+        .execute()
+
+
+def test_decimal(trino_connection):
+    SqlTest(trino_connection) \
+        .add_field(sql="CAST(null AS DECIMAL)", python=None) \
+        .add_field(sql="CAST(null AS DECIMAL(38,0))", python=None) \
+        .add_field(sql="DECIMAL '10.3'", python=Decimal('10.3')) \
+        .add_field(sql="CAST('0.123456789123456789' AS DECIMAL(18,18))", python=Decimal('0.123456789123456789')) \
+        .add_field(sql="CAST(null AS DECIMAL(18,18))", python=None) \
+        .add_field(sql="CAST('234.123456789123456789' AS DECIMAL(18,4))", python=Decimal('234.1235')) \
+        .add_field(sql="CAST('10.3' AS DECIMAL(38,1))", python=Decimal('10.3')) \
+        .add_field(sql="CAST('0.123456789123456789' AS DECIMAL(18,2))", python=Decimal('0.12')) \
+        .execute()
+
+
+def test_varchar(trino_connection):
+    SqlTest(trino_connection) \
+        .add_field(sql="'aaa'", python='aaa') \
+        .add_field(sql="U&'Hello winter \2603 !'", python='Hello winter °3 !') \
+        .add_field(sql="CAST(null AS VARCHAR)", python=None) \
+        .add_field(sql="CAST('bbb' AS VARCHAR(1))", python='b') \
+        .add_field(sql="CAST(null AS VARCHAR(1))", python=None) \
+        .execute()
+
+
+def test_char(trino_connection):
+    SqlTest(trino_connection) \
+        .add_field(sql="CAST('ccc' AS CHAR)", python='c') \
+        .add_field(sql="CAST(null AS CHAR)", python=None) \
+        .add_field(sql="CAST('ddd' AS CHAR(1))", python='d') \
+        .add_field(sql="CAST('😂' AS CHAR(1))", python='😂') \
+        .add_field(sql="CAST(null AS CHAR(1))", python=None) \
+        .execute()
+
+
+def test_varbinary(trino_connection):
+    SqlTest(trino_connection) \
+        .add_field(sql="X'65683F'", python='ZWg/') \
```

Comment:
also test some varbinary input which isn't a valid UTF-8 string.

Reply:
Actually it doesn't seem to work. SELECT CAST('😂😂😂😂😂😂' AS VARBINARY) returns `8J+YgvCfmILwn5iC8J+YgvCfmILwn5iC` in Starburst itself.

Reply:
It's expected. Varbinary returns raw bytes. However if you run from_utf8 on that varbinary output then it should give you back the string.

```
trino> SELECT from_utf8(CAST('😂😂😂😂😂😂' AS VARBINARY));
    _col0
--------------
 😂😂😂😂😂😂
(1 row)
```

Reply:
A test like `sql="X'0001020304050607080DF9367AA7000000'", python='<whatever bag of bytes we get back>'` should be fine.

Reply:
Done